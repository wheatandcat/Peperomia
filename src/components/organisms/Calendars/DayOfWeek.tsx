import React from 'react';
import { Text, View, StyleSheet } from 'react-native';
import theme from 'config/theme';

type Props = {};

const DayOfWeek: React.FC<Props> = () => {
  return (
    <View style={styles.weekNameContainer}>
      <Text style={[styles.weekText, { color: theme().color.error.main }]}>
        日
      </Text>
      <Text style={styles.weekText}>月</Text>
      <Text style={styles.weekText}>火</Text>
      <Text style={styles.weekText}>水</Text>
      <Text style={styles.weekText}>木</Text>
      <Text style={styles.weekText}>金</Text>
      <Text style={[styles.weekText, { color: theme().color.accent1.main }]}>
        土
      </Text>
    </View>
  );
};

export default DayOfWeek;

const styles = StyleSheet.create({
  weekNameContainer: {
    marginTop: theme().space(2),
    flexDirection: 'row',
    justifyContent: 'space-around',
    marginHorizontal: theme().space(2),
  },
  weekText: {
    paddingVertical: theme().space(3),
    textAlign: 'center',
    alignItems: 'center',
    fontSize: 18,
    fontWeight: '600',
  },
});
