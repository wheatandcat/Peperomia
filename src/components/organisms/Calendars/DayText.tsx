import React from 'react';
import {
  Text,
  View,
  Dimensions,
  StyleSheet,
  Platform,
  StatusBar,
} from 'react-native';
import { getStatusBarHeight } from 'react-native-status-bar-height';
import Color from 'color';
import { isTablet } from 'lib/responsive';
import theme from 'config/theme';
import { getWeekCount } from 'lib/calendar';

const width = Dimensions.get('window').width;
const top =
  Platform.OS === 'android'
    ? StatusBar.currentHeight || 0
    : getStatusBarHeight() || 0;
const footer = 120;
const header = 100;
const height = Dimensions.get('window').height - top - header - footer;

type Props = {
  currentDate: string;
  color: string;
  day: number;
};

const DayText: React.FC<Props> = (props) => {
  const backgroundColor = Color(props.color).fade(0.8).toString();
  const rows = getWeekCount(props.currentDate);

  return (
    <View
      style={[
        styles.itemContainer,
        {
          backgroundColor,
          height: height / rows,
          borderColor: props.color,
        },
      ]}
    >
      <Text style={isTablet ? styles.dayTextForWide : styles.dayText}>
        {props.day}
      </Text>
    </View>
  );
};

export default DayText;

const styles = StyleSheet.create({
  itemContainer: {
    width: width / 7,
    borderTopWidth: 0.5,
    borderRightWidth: 0.5,
    borderLeftWidth: 0.5,
    borderTopColor: theme().color.gray,
    borderBottomColor: theme().color.gray,
    paddingTop: 15,
    alignItems: 'center',
  },
  dayText: {
    textAlign: 'center',
    color: theme().color.gray,
    fontSize: 18,
    fontWeight: '600',
  },
  dayTextForWide: {
    textAlign: 'center',
    color: theme().color.gray,
    fontSize: 26,
    fontWeight: '600',
  },
});
