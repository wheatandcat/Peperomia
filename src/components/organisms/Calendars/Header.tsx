import React from 'react';
import { Text, View, TouchableOpacity, StyleSheet } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import theme from 'config/theme';

type Props = {
  onPrevMonth: () => void;
  onNextMonth: () => void;
  date: string;
};

const Header: React.FC<Props> = (props) => {
  return (
    <View style={styles.headerContainer}>
      <TouchableOpacity onPress={props.onPrevMonth}>
        <MaterialCommunityIcons
          name="chevron-left"
          size={30}
          color={theme().color.primary.main}
        />
      </TouchableOpacity>

      <View style={styles.yearContainer}>
        <Text style={styles.year}>{props.date}</Text>
      </View>

      <TouchableOpacity onPress={props.onNextMonth}>
        <MaterialCommunityIcons
          name="chevron-right"
          size={30}
          color={theme().color.primary.main}
        />
      </TouchableOpacity>
    </View>
  );
};

export default Header;

const styles = StyleSheet.create({
  headerContainer: {
    justifyContent: 'space-between',
    alignItems: 'center',
    flexDirection: 'row',
  },
  yearContainer: {
    paddingVertical: theme().space(3),
    justifyContent: 'center',
  },
  year: {
    textAlign: 'center',
    color: theme().color.darkGray,
    fontSize: 25,
    fontWeight: '600',
  },
});
