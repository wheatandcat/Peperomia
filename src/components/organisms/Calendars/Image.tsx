import React from 'react';
import {
  Text,
  View,
  Dimensions,
  StyleSheet,
  Platform,
  StatusBar,
} from 'react-native';
import { getStatusBarHeight } from 'react-native-status-bar-height';
import { IconImage } from 'components/atoms';
import { isTablet } from 'lib/responsive';
import { KINDS } from 'peperomia-util';
import theme from 'config/theme';
import { getWeekCount } from 'lib/calendar';

const width = Dimensions.get('window').width;
const top =
  Platform.OS === 'android'
    ? StatusBar.currentHeight || 0
    : getStatusBarHeight() || 0;
const footer = 120;
const header = 100;
const height = Dimensions.get('window').height - top - header - footer;

type Props = {
  currentDate: string;
  kind: string;
  day: string;
};

export default (props: Props) => {
  const config = KINDS[props.kind];
  const rows = getWeekCount(props.currentDate);

  return (
    <View
      style={[
        styles.itemContainer,
        { backgroundColor: config.backgroundColor, height: height / rows },
      ]}
    >
      <Text style={isTablet ? styles.dayTextForWide : styles.dayText}>
        {props.day}
      </Text>
      <View style={styles.imageContainer}>
        <IconImage
          src={config.src}
          name={config.name}
          opacity={0.8}
          size={isTablet ? 60 : 30}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  itemContainer: {
    width: width / 7,
    borderTopWidth: 0.5,
    borderRightWidth: 0.5,
    borderLeftWidth: 0.5,
    borderColor: theme().color.gray,
    paddingTop: 15,
    alignItems: 'center',
  },
  dayText: {
    textAlign: 'center',
    color: theme().color.black,
    fontSize: 12,
    fontWeight: '600',
    paddingBottom: 7,
  },
  dayTextForWide: {
    textAlign: 'center',
    color: theme().color.black,
    fontSize: 20,
    fontWeight: '600',
    paddingBottom: 7,
  },
  imageContainer: {
    alignItems: 'center',
    justifyContent: 'center',
    height: '55%',
  },
});
