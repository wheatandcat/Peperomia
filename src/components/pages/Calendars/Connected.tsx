import React, { memo, useCallback } from 'react';
import dayjs from 'dayjs';
import advancedFormat from 'dayjs/plugin/advancedFormat';
import 'dayjs/locale/ja';
import { ContextProps as CalendarsContextProps } from 'containers/Calendars';
import { Props as IndexProps } from './';
import Page from './Page';

dayjs.extend(advancedFormat);

type Props = IndexProps &
  Pick<CalendarsContextProps, 'calendars' | 'setDate'> & {
    loading: boolean;
  };

export type ConnectedType = {
  onCreate: (date: string) => void;
  onCalendar: (date: string) => void;
  onSchedule: (id: string | number, title: string) => void;
};

const Connected: React.FC<Props> = memo((props) => {
  const onCreate = useCallback(
    (date: string) => {
      props.navigation.navigate('CreatePlan', {
        date,
      });
    },
    [props.navigation]
  );

  const onSchedule = useCallback(
    (id: string | number, title: string) => {
      props.navigation.navigate('Schedule', { itemId: id, title });
    },
    [props.navigation]
  );

  const onCalendar = useCallback(
    (date: string) => {
      props.navigation.navigate('Calendar', {
        date: dayjs(date).format('YYYY-MM-DDT00:00:00'),
      });
    },
    [props.navigation]
  );

  return (
    <Page
      calendars={props.calendars}
      setDate={props.setDate}
      loading={props.loading}
      onCreate={onCreate}
      onSchedule={onSchedule}
      onCalendar={onCalendar}
    />
  );
});

export default Connected;
