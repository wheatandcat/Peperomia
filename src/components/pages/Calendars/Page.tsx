import React, { Component } from 'react';
import {
  Text,
  View,
  TouchableOpacity,
  ActivityIndicator,
  Animated,
  SafeAreaView,
  ScrollView,
  StatusBar,
} from 'react-native';
import { Calendar, LocaleConfig, CalendarTheme } from 'react-native-calendars';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import GestureRecognizer from 'react-native-swipe-gestures';
import EStyleSheet from 'react-native-extended-stylesheet';
import dayjs from 'dayjs';
import advancedFormat from 'dayjs/plugin/advancedFormat';
import 'dayjs/locale/ja';
import { SelectCalendar } from 'domain/calendar';
import theme from 'config/theme';
import ImageDay from 'components/organisms/Calendars/Image';
import DayText from 'components/organisms/Calendars/DayText';
import GlobalStyles from '../../../GlobalStyles';
import { ConnectedType } from './Connected';

dayjs.extend(advancedFormat);

const AnimatedSafeAreaView = Animated.createAnimatedComponent(SafeAreaView);

LocaleConfig.locales.jp = {
  monthNames: [
    '1月',
    '2月',
    '3月',
    '4月',
    '5月',
    '6月',
    '7月',
    '8月',
    '9月',
    '10月',
    '11月',
    '12月',
  ],
  monthNamesShort: [
    '1月',
    '2月',
    '3月',
    '4月',
    '5月',
    '6月',
    '7月',
    '8月',
    '9月',
    '10月',
    '11月',
    '12月',
  ],
  dayNames: ['日', '月', '火', '水', '木', '金', '土'],
  dayNamesShort: ['日', '月', '火', '水', '木', '金', '土'],
};
LocaleConfig.defaultLocale = 'jp';

type Props = ConnectedType & {
  loading: boolean;
  calendars: SelectCalendar[];
};

type State = {
  currentDate: string;
  count: number;
  backgroundColor: any;
};

const backgroundColors = [
  '#FDDACB',
  '#FBEDF2',
  '#DCF6FA',
  '#fdeff2',
  '#FFE8AE',
  '#7BB07D',
  '#a0d8ef',
  '#f2f2b0',
  theme().color.beige,
  '#E3E3FE',
  '#FFE7D0',
  '#BFD3B7',
];

const config = {
  velocityThreshold: 0.3,
  directionalOffsetThreshold: 80,
};

export default class extends Component<Props, State> {
  state = {
    currentDate: dayjs().format('YYYY-MM-01'),
    count: 0,
    backgroundColor: new Animated.Value(dayjs().month()),
  };

  calendar: any;
  componentDidMount() {
    const currentMonth = dayjs(this.state.currentDate).month();

    Animated.timing(this.state.backgroundColor, {
      toValue: currentMonth,
      useNativeDriver: false,
    }).start();
  }

  onNextMonth = () => {
    const count = this.state.count + 1;
    const currentMonth = dayjs(this.state.currentDate)
      .add(count, 'month')
      .month();

    this.setState({
      count,
    });

    if (currentMonth === 0) {
      Animated.timing(this.state.backgroundColor, {
        toValue: 12,
        useNativeDriver: false,
      }).start((props) => {
        if (props.finished) {
          this.setState({
            backgroundColor: new Animated.Value(0),
          });
        }
      });
    } else {
      Animated.timing(this.state.backgroundColor, {
        toValue: currentMonth,
        useNativeDriver: false,
      }).start();
    }
    this.calendar.addMonth(1);
  };

  onPrevMonth = () => {
    const count = this.state.count - 1;
    const currentMonth = dayjs(this.state.currentDate)
      .add(count, 'month')
      .month();

    this.setState({
      count,
    });

    if (currentMonth === 11) {
      Animated.timing(this.state.backgroundColor, {
        toValue: -1,
        useNativeDriver: false,
      }).start((props) => {
        if (props.finished) {
          this.setState({
            backgroundColor: new Animated.Value(12),
          });
        }
      });
    } else {
      Animated.timing(this.state.backgroundColor, {
        toValue: currentMonth,
        useNativeDriver: false,
      }).start();
    }
    this.calendar.addMonth(-1);
  };

  render() {
    const backgroundColor = this.state.backgroundColor.interpolate({
      inputRange: [-1, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12],
      outputRange: [
        backgroundColors[11],
        ...backgroundColors,
        backgroundColors[0],
      ],
    });

    const currentMonth = dayjs(this.state.currentDate)
      .add(this.state.count, 'month')
      .month();

    const currentDate = dayjs(this.state.currentDate)
      .add(this.state.count, 'month')
      .format('YYYY-MM-DD');

    const animationStyle = {
      backgroundColor,
    };

    return (
      <ScrollView
        style={[
          styles.scroll,
          {
            backgroundColor: backgroundColors[currentMonth],
          },
        ]}
      >
        <StatusBar
          backgroundColor={theme().color.white}
          barStyle={'dark-content'}
        />
        <AnimatedSafeAreaView
          style={[GlobalStyles.droidSafeArea, styles.safeArea, animationStyle]}
        />
        <GestureRecognizer
          onSwipeLeft={this.onNextMonth}
          onSwipeRight={this.onPrevMonth}
          config={config}
        >
          <AnimatedSafeAreaView style={[styles.root, animationStyle]}>
            <View style={styles.headerContainer}>
              <TouchableOpacity onPress={this.onPrevMonth}>
                <MaterialCommunityIcons
                  name="chevron-left"
                  size={30}
                  color={theme().color.main}
                />
              </TouchableOpacity>

              <View style={styles.yearContainer}>
                <Text style={styles.year}>
                  {dayjs(this.state.currentDate)
                    .add(this.state.count, 'month')
                    .format('YYYY年MM月')}
                </Text>
              </View>

              <TouchableOpacity onPress={this.onNextMonth}>
                <MaterialCommunityIcons
                  name="chevron-right"
                  size={30}
                  color={theme().color.main}
                />
              </TouchableOpacity>
            </View>

            <View style={styles.weekNameContainer}>
              <Text style={[styles.weekText, { color: theme().color.red }]}>
                日
              </Text>
              <Text style={styles.weekText}>月</Text>
              <Text style={styles.weekText}>火</Text>
              <Text style={styles.weekText}>水</Text>
              <Text style={styles.weekText}>木</Text>
              <Text style={styles.weekText}>金</Text>
              <Text style={[styles.weekText, { color: theme().color.sky }]}>
                土
              </Text>
            </View>
            {this.props.loading ? (
              <ActivityIndicator size="large" color={theme().mode.text} />
            ) : (
              <Calendar
                ref={(ref) => (this.calendar = ref)}
                style={{
                  backgroundColor: backgroundColors[currentMonth],
                }}
                monthFormat={''}
                hideDayNames
                hideArrows
                theme={calendarTheme}
                dayComponent={({ date }) => {
                  const schedule = this.props.calendars.find(
                    (item) => item.date === date.dateString
                  );

                  if (schedule) {
                    return (
                      <TouchableOpacity
                        onPress={() =>
                          this.props.onSchedule(schedule.itemId, schedule.title)
                        }
                      >
                        <ImageDay
                          currentDate={currentDate}
                          kind={schedule.kind}
                          day={String(date.day)}
                        />
                      </TouchableOpacity>
                    );
                  }

                  return (
                    <TouchableOpacity
                      onPress={() => this.props.onCreate(date.dateString)}
                    >
                      <DayText
                        currentDate={currentDate}
                        color={backgroundColors[currentMonth]}
                        day={date.day}
                      />
                    </TouchableOpacity>
                  );
                }}
              />
            )}
            <View style={styles.calendarBottom} />
          </AnimatedSafeAreaView>
        </GestureRecognizer>
      </ScrollView>
    );
  }
}

const calendarTheme: CalendarTheme = {
  'stylesheet.calendar.main': {
    week: {
      marginTop: 0,
      flexDirection: 'row',
      justifyContent: 'space-between',
    },
  },
  textMonthFontSize: 18,
  monthTextColor: '#000',
  textMonthFontWeight: '600',
  'stylesheet.calendar.header': {
    header: {
      height: 0,
    },
    dayHeader: {
      fontWeight: '600',
      paddingBottom: 10,
      color: '#000',
    },
  },
};

const styles = EStyleSheet.create({
  root: {
    flex: 1,
    height: '100%',
    width: '100%',
  },
  headerContainer: {
    justifyContent: 'space-between',
    alignItems: 'center',
    flexDirection: 'row',
  },
  yearContainer: {
    paddingVertical: 20,
    justifyContent: 'center',
  },
  year: {
    textAlign: 'center',
    color: theme().color.darkGray,
    fontSize: 25,
    fontWeight: '600',
  },
  weekNameContainer: {
    marginTop: 7,
    flexDirection: 'row',
    justifyContent: 'space-around',
    marginHorizontal: 4,
  },
  weekText: {
    paddingVertical: 12,
    textAlign: 'center',
    alignItems: 'center',
    fontSize: 18,
    fontWeight: '600',
  },
  calendarBottom: {
    borderColor: 'gray',
    marginHorizontal: 4,
    borderBottomWidth: 1,
  },
  scroll: {
    height: '100%',
    width: '100%',
  },
  safeArea: {
    flex: 0,
  },
});
