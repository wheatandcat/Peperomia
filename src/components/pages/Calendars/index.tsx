import React, { memo } from 'react';
import { RouteProp } from '@react-navigation/native';
import {
  createStackNavigator,
  StackNavigationProp,
} from '@react-navigation/stack';
import { RootStackParamList } from 'lib/navigation';
import { useCalendars } from 'containers/Calendars';
import Calendar from 'components/pages/Calendar/index';
import ItemDetail from 'components/pages/ItemDetail';
import AddItemDetail from 'components/pages/AddItemDetail/index';
import EditItemDetail from 'components/pages/EditItemDetail/index';
import { CreateCalendar } from 'components/pages/CreateCalendar/index';
import Icons, { IconsNavigationOptions } from 'components/pages/Icons';
import Connected from './Connected';

type CalendarsScreenNavigationProp = StackNavigationProp<
  RootStackParamList,
  'Calendars'
>;
type ScreenRouteProp = RouteProp<RootStackParamList, 'Calendars'>;

export type Props = {
  navigation: CalendarsScreenNavigationProp;
  route: ScreenRouteProp;
};

const Calendars: React.FC<Props> = memo((props) => {
  const { calendars, loadingCalendars, setDate } = useCalendars();

  return (
    <Connected
      {...props}
      calendars={calendars}
      loading={loadingCalendars}
      setDate={setDate}
    />
  );
});

const Stack = createStackNavigator<RootStackParamList>();

const RootStack = () => {
  return (
    <Stack.Navigator initialRouteName="Calendars" mode="modal">
      <Stack.Screen
        name="Calendars"
        component={Calendars}
        options={{ header: () => null }}
      />
      <Stack.Screen
        name="Calendar"
        component={Calendar}
        options={{ headerShown: false }}
      />
      <Stack.Screen
        name="CreateCalendar"
        component={CreateCalendar}
        options={{ header: () => null }}
      />
      <Stack.Screen
        name="ItemDetail"
        component={ItemDetail}
        options={{ header: () => null }}
      />
      <Stack.Screen
        name="AddItemDetail"
        component={AddItemDetail}
        options={{ header: () => null }}
      />
      <Stack.Screen
        name="EditItemDetail"
        component={EditItemDetail}
        options={{ header: () => null }}
      />
      <Stack.Screen
        name="Icons"
        component={Icons}
        options={IconsNavigationOptions}
      />
    </Stack.Navigator>
  );
};

export default RootStack;
