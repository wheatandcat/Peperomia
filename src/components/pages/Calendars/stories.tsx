import React from 'react';
import { View, StyleSheet } from 'react-native';
import { storiesOf } from '@storybook/react-native';
import { KIND_PARK, KIND_ART_MUSEUM, KIND_COFFEE } from 'peperomia-util';
import { mockFn, StackNavigator } from 'storyBookUtils';
import Page from './Page';

const calendars = [
  {
    id: 'a',
    date: '2019-10-03',
    item: {
      id: 'a',
      kind: KIND_COFFEE,
    },
  },
  {
    id: 'b',
    date: '2019-10-14',
    item: {
      id: 'b',
      kind: KIND_PARK,
    },
  },
  {
    id: 'b',
    date: '2019-10-24',
    item: {
      id: 'b',
      kind: KIND_ART_MUSEUM,
    },
  },
];

const Screen = () => {
  return (
    <View style={styles.root}>
      <Page
        calendars={calendars}
        onCalendar={mockFn('onCalendar')}
        onCreate={mockFn('onCreate')}
        setDate={mockFn('setDate')}
        loading={false}
      />
    </View>
  );
};

storiesOf('pages', module).add('Calendars', () => (
  <StackNavigator screen={Screen} />
));

const styles = StyleSheet.create({
  root: {
    paddingTop: 0,
  },
});
