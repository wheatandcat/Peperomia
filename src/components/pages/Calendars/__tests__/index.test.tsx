import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import * as ContainerCalendars from 'containers/Calendars';
import { calendarsMockData } from '__mockData__/calendar.ts';
import Calendars from '../';

const propsData = (): any =>
  ({
    navigation: {
      setParams: jest.fn(),
      navigate: jest.fn(),
    },
    route: {
      params: {},
    },
  } as any);

describe('components/pages/Calendars/index.tsx', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    jest.spyOn(ContainerCalendars, 'useCalendars').mockImplementation(
      () =>
        ({
          calendars: calendarsMockData(),
          loadingCalendars: false,
          setDate: jest.fn(),
        } as any)
    );

    wrapper = shallow(<Calendars {...propsData()} />);
  });

  it('正常にrenderすること', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
