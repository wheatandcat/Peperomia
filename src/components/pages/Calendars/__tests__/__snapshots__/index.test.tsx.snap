// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/Calendars/index.tsx 正常にrenderすること 1`] = `
<StackNavigator
  initialRouteName="Calendars"
  mode="modal"
>
  <Screen
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": [Function],
      }
    }
    name="Calendars"
    options={
      Object {
        "header": [Function],
      }
    }
  />
  <Screen
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": [Function],
      }
    }
    name="Calendar"
    options={
      Object {
        "headerShown": false,
      }
    }
  />
  <Screen
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": [Function],
      }
    }
    name="CreateCalendar"
    options={
      Object {
        "header": [Function],
      }
    }
  />
  <Screen
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": [Function],
      }
    }
    name="ItemDetail"
    options={
      Object {
        "header": [Function],
      }
    }
  />
  <Screen
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": [Function],
      }
    }
    name="AddItemDetail"
    options={
      Object {
        "header": [Function],
      }
    }
  />
  <Screen
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": [Function],
      }
    }
    name="EditItemDetail"
    options={
      Object {
        "header": [Function],
      }
    }
  />
  <Screen
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": [Function],
      }
    }
    name="Icons"
    options={[Function]}
  />
</StackNavigator>
`;
